### PR TITLE
Allow exec to create and edit invoices and contracts, fix timecard default dropdown option

### PR DESCRIPTION
**Pull request by wjiang42**
Resolve https://github.com/ABTech/tracker/issues/521:

- Ability.rb grants status permission to >= execs
- Invoice edit view: status dropdown is determined by manage permission and status permission
- Event view: removed ability to change invoice status from event page as this made it too easy to accidentally change an invoice status type

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -80,14 +80,13 @@ def initialize(member)
...
-      can :readprice, Invoice
-      can :create, Invoice
-      can :update, Invoice, :status => Invoice::Invoice_Status_Group_Exec
-      can :email, Invoice, :status => Invoice::Invoice_Status_Group_Exec
-      can :read, InvoiceItem
-      can :read, InvoiceContact
+      can :read, :finance_event_url
+      can :manage, Invoice
+      can :manage, InvoiceItem
+      can :manage, InvoiceContact
```
Comment (DaAwesomeP) [inline app/models/ability.rb:86]: `InvoiceContact` is who in SLICE gets sent invoices. Are you certain that should be all of exec people or just finance (Tracker managers)?

File: app/models/ability.rb [hunk 2]
```diff
@@ -80,14 +80,13 @@ def initialize(member)
...
-      can :read, Invoice
-      can :readprice, Invoice
-      can :create, Invoice
-      can :update, Invoice, :status => Invoice::Invoice_Status_Group_Exec
-      can :email, Invoice, :status => Invoice::Invoice_Status_Group_Exec
-      can :read, InvoiceItem
-      can :read, InvoiceContact
+      can :read, :finance_event_url
+      can :manage, Invoice
+      can :manage, InvoiceItem
```
Comment (DaAwesomeP) [inline app/models/ability.rb:85]: `InvoiceItem` are the dropdown items when building a quote. Are you certain that should be all of exec people or just finance (Tracker managers)?

File: app/models/ability.rb [hunk 3]
```diff
@@ -80,14 +80,13 @@ def initialize(member)
...
       # Almost read only finance
-      can :read, :finance
-      can :read, Invoice
-      can :readprice, Invoice
-      can :create, Invoice
-      can :update, Invoice, :status => Invoice::Invoice_Status_Group_Exec
-      can :email, Invoice, :status => Invoice::Invoice_Status_Group_Exec
-      can :read, InvoiceItem
-      can :read, InvoiceContact
+      can :read, :finance_event_url
```
Comment (DaAwesomeP) [inline app/models/ability.rb:83]: Why change this to `finance_event_url`? I believe this will cause the Finance tab to disappear?

File: app/views/invoices/_form.html.erb [hunk 4]
```diff
@@ -27,10 +27,14 @@
 
   <tr>
     <td class="subheading"><%= f.label :status, "Status" %>:</td>
-    <% if can? :status, f.object %>
+    <% if current_member.is_at_least? :tracker_management %>
```
Comment (DaAwesomeP) [inline app/views/invoices/_form.html.erb:30]: Use abilities; don't hard-code in member tiers because it makes it harder to find/manage abilities across the whole site when changing things later. Instead of setting member groups here you can just give more people the `:status` ability.

This is also inconsistent with the permissions you have set in `ability.rb`. This will show/hide a form but under the hood you have given all of exec all permissions on invoices regardless.

File: app/views/events/show.html.erb [hunk 5]
```diff
@@ -100,15 +100,7 @@
                     <%= link_to invoice.display_title, invoice %>
                 </td>
                 <td>
-                  <% if can? :status, invoice %>
```
Comment (DaAwesomeP) [inline app/views/events/show.html.erb:103]: To be thorough, leave this `if` statement. That way if permissions get changed in the future the box will still be show/hidden properly and it won't break the `:status` permission when used.

File: app/views/invoices/_form.html.erb [hunk 6]
```diff
@@ -27,10 +27,14 @@
...
       <td><%= f.select :status, Invoice::Invoice_Status_Group_All %></td>
-    <% elsif f.object.new_record? %>
-      <td><%= f.select :status, Invoice::Invoice_Status_Group_Exec %></td>
+    <% elsif current_member.is_at_least? :exec%>
+      <% if f.object.status == "Received" || f.object.status == "Loan Agreement" %>
```
Comment (DaAwesomeP) [inline app/views/invoices/_form.html.erb:33]: Don't case on specific statuses here if you can use the `Invoice_Status_Group_All` and `Invoice_Status_Group_Exec` parts already defined.

This form filter is also not consistent with what you changed in `ability.rb`, which gives exec access to change all.

**Review by DaAwesomeP — CHANGES_REQUESTED** [review]
There are some conflicting changes here with permissions granted vs what the forms allow.

File: app/views/timecard_entries/index.html.erb
Comment (DaAwesomeP) [inline app/views/timecard_entries/index.html.erb:1]: This change looks unrelated. Did you mean to push to another branch?

File: app/views/timecards/show.html.erb
Comment (DaAwesomeP) [inline app/views/timecards/show.html.erb:1]: This change looks unrelated. Did you mean to push to another branch?

**Review by DaAwesomeP — CHANGES_REQUESTED** [review]

File: app/views/timecard_entries/index.html.erb
Comment (wjiang42) [reply app/views/timecard_entries/index.html.erb:1]: Yes, I was working on a fix for #519 and forgot that committing to master would add it to the PR. 

All I did was use the same dummy selection option from the runcrew roles dropdown in the events new form. I wasn't able to test it locally because of issues with displaying timecards, but since it's broken with or without the change and it should be harmless I'd like to push it if possible.

File: app/views/timecards/show.html.erb
Comment (wjiang42) [reply app/views/timecards/show.html.erb:1]: ^
